S3Upload.java

```diff
@@ -0,0 +1,58 @@
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+
+import java.io.*;
+
+/**
+ * Upload a file to an Amazon S3 bucket.
+ * <p>
+ * This code expects that you have AWS credentials set up per:
+ * http://docs.aws.amazon.com/java-sdk/latest/developer-guide/setup-credentials.html
+ */
+public class S3Upload {
+    public static void main(String[] args) {
+        String bucket_name = "<FMI1>";
+        String file_path = "<FMI2>";
+        String key_name = "<FMI3>";
+        String fileContents = readFileContents();
+
+        System.out.format("Uploading %s to S3 bucket %s...\n", file_path, bucket_name);
+        final AmazonS3 s3 = AmazonS3ClientBuilder.standard().withRegion(Regions.DEFAULT_REGION).build();
+        try {
+            s3.putObject(bucket_name, key_name, new File(fileContents));
+        } catch (AmazonServiceException e) {
+            System.err.println(e.getErrorMessage());
+            System.exit(1);
+        }
+        System.out.println("Done!");
+    }
+
+    private static String readFileContents() {
+        BufferedReader reader = null;
+        String fileContents = "";
+        try {
+           // reader = new BufferedReader(new FileReader(
+             //       "C:\\<FMI2>"));
+            FileReader fr = new FileReader(
+                    "C:\\<FMI2>");
+            reader = new BufferedReader(fr);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *reader*. The resource is referenced at lines 40, 43. The resource is closed at lines 48, 49. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *reader* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).